Foundation/NSString.swift

```diff
@@ -343,8 +343,18 @@ open class NSString : NSObject, NSCopying, NSMutableCopying, NSSecureCoding, NSC
 
 extension NSString {
     public func getCharacters(_ buffer: UnsafeMutablePointer<unichar>, range: NSRange) {
-        for idx in 0..<range.length {
-            buffer[idx] = character(at: idx + range.location)
+        if type(of: self) == NSString.self || type(of: self) == NSMutableString.self {
```

Comment:
How much does this type check slow things down? I suppose anything is better than `character(at:)`...

Reply:
There's a lot of things we could do to improve this further. My goal here is just to get it from n^2 to n so that it's not a huge problem. If we find important uses of it we can sink more time into it.